src/rules/extensions.ts

```diff
@@ -188,6 +193,20 @@ function computeOverrideAction(
   }
 }
 
+/**
+ * Replaces the import path in a source string with a new import path.
+ *
+ * @param source - The original source string containing the import statement.
+ * @param importPath - The new import path to replace the existing one.
+ * @returns The updated source string with the replaced import path.
+ */
+function replaceImportPath(source: string, importPath: string) {
```

Comment:
@stephenjason89 You also need to change your `fix` implementation for https://github.com/import-js/eslint-plugin-import/pull/3177. `JSON.stringify` is not correct for single quoted imports.